content/courses/connecting-to-offchain-data/oracles.md

```diff
@@ -1179,43 +1220,55 @@ export const solUsedSwitchboardFeed = new anchor.web3.PublicKey(
   "GvDMxPzN1sCj7L26YDK2HnMRXEQmQ2aemov8YBtPS7vR",
 );
 
-describe("burry-escrow", () => {
-  // Configure the client to use the local cluster.
+describe("Burry Escrow", () => {
   anchor.setProvider(anchor.AnchorProvider.env());
   const provider = anchor.AnchorProvider.env();
   const program = anchor.workspace.BurryEscrow as Program<BurryEscrow>;
   const payer = (provider.wallet as AnchorWallet).payer;
 
-  it("Create Burry Escrow Below Price", async () => {
-    // fetch switchboard devnet program object
-    const switchboardProgram = await SwitchboardProgram.load(
+  const ESCROW_IDENTIFIER = "MICHAEL BURRY";
+  const SOL_PRICE_DECREMENT = 10;
+  const SOL_PRICE_INCREMENT = 10;
+  const AMOUNT_TO_LOCK_UP = new anchor.BN(100);
+
+  async function fetchSwitchboardProgram() {
+    return await SwitchboardProgram.load(
       "devnet",
       new anchor.web3.Connection("https://api.devnet.solana.com"),
       payer,
     );
-    const aggregatorAccount = new AggregatorAccount(
-      switchboardProgram,
-      solUsedSwitchboardFeed,
-    );
+  }
 
-    // derive escrow state account
-    const [escrowState] = await anchor.web3.PublicKey.findProgramAddressSync(
-      [Buffer.from("MICHAEL BURRY"), payer.publicKey.toBuffer()],
+  async function deriveEscrowState() {
+    return await anchor.web3.PublicKey.findProgramAddressSync(
+      [Buffer.from(ESCROW_IDENTIFIER), payer.publicKey.toBuffer()],
       program.programId,
     );
+  }
 
-    // fetch latest SOL price
+  async function fetchSolPrice(aggregatorAccount: AggregatorAccount) {
     const solPrice: Big | null = await aggregatorAccount.fetchLatestValue();
     if (solPrice === null) {
       throw new Error("Aggregator holds no value");
     }
-    const failUnlockPrice = solPrice.minus(10).toNumber();
-    const amountToLockUp = new anchor.BN(100);
+    return solPrice;
+  }
+
+  it("Create Burry Escrow Below Price", async () => {
```

Comment:
`Creates`. See https://github.com/solana-foundation/developer-content/blob/main/CONTRIBUTING.md#jsts re: BDD. Do this for all tests.